clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -181,13 +182,24 @@ public static boolean validateRequiredResourceLabels(Map<String, String> metadat
         return validateResourceLabel(metadata, MetricsContext.NAMESPACE);
     }
 
-    public static CompressionType preferredCompressionType(List<CompressionType> acceptedCompressionTypes) {
-        if (acceptedCompressionTypes != null && !acceptedCompressionTypes.isEmpty()) {
-            // Broker is providing the compression types in order of preference. Grab the
-            // first one.
-            return acceptedCompressionTypes.get(0);
-        }
-        return CompressionType.NONE;
+    /**
+     * Determines the preferred compression type from broker-accepted types, avoiding unsupported ones.
+     * 
+     * @param acceptedCompressionTypes the list of compression types accepted by the broker in order 
+     *                                of preference (must not be null, use empty list if no compression is accepted)
+     * @param unsupportedCompressionTypes the set of compression types that should be avoided due to 
+     *                                   missing libraries or previous failures (must not be null)
+     * @return the preferred compression type to use, or {@link CompressionType#NONE} if no acceptable
+     *         compression type is available
+     * @throws NullPointerException if acceptedCompressionTypes or unsupportedCompressionTypes is null
```

Comment:
Having NPE in @throws and being documented seem a bit inappropriate. You have already mentioned that non-null values are expected for these params.